application-admintools-test/application-admintools-test-docker/src/test/it/com/xwiki/admintools/test/ui/AdminToolsIT.java

```diff
@@ -0,0 +1,300 @@
+        new DocumentReference("xwiki", Arrays.asList("AdminTools", "Code"), "Configuration");
+
+    private static final String ADMINTOOLS_CONFIGURATION_CLASSNAME = "AdminTools.Code.ConfigurationClass";
+
+    private static final String BACKEND_SECTION_VIEW_LAST_LOGS_MODAL_ID = "#configurationViewLastNLinesModal";
+
+    private static final String DOWNLOAD_FILES_MODAL_ID = "#downloadFilesModal";
+
+    private static final String PASSWORD = "pass";
+
+    private static boolean isSupportedServer = true;
+
+    private final List<String> supportedDatabases =
+        List.of("mysql", "hsql", "hsqldb", "mariadb", "postgresql", "oracle");
+
+    private final List<String> excludedLines =
+        List.of("environment.permanentDirectory", "rendering.linkLabelFormat", "core.defaultDocumentSyntax", "MISC");
+
+    @BeforeAll
+    static void setUp(TestUtils setup, TestConfiguration testConfiguration)
+    {
+        setup.createUser(USER_NAME, PASSWORD, setup.getURLToNonExistentPage(), "first_name", "Jon", "last_name",
+            "Snow");
+        String serverType = testConfiguration.getServletEngine().name();
+        if (!supportedServers.contains(serverType)) {
+            isSupportedServer = false;
+            AdminToolsHomePage page = AdminToolsHomePage.gotoPage();
+            setup.gotoPage(page.getPageURL());
+            AdminToolsViewPage webHomePage = new AdminToolsViewPage();
+            assertEquals(0, webHomePage.getDashboardElements().size());
+            List<WebElement> warningElements = webHomePage.getWarningElements();
+            assertEquals(2, warningElements.size());
+        }
+
+        // By default the minimal distribution used for the tests doesn't have any rights setup. Let's create an Admin
+        // user part of the Admin Group and make sure that this Admin Group has admin rights in the wiki. We could also
+        // have given that Admin user the admin right directly but the solution we chose is closer to the XS
+        // distribution.
+        setup.loginAsSuperAdmin();
+        setup.setGlobalRights("XWiki.XWikiAdminGroup", "", "admin", true);
+        setup.createAdminUser();
+        setup.loginAsAdmin();
+    }
+
+    @BeforeEach
+    void goToPage(TestUtils testUtils)
+    {
+        AdminToolsHomePage page = AdminToolsHomePage.gotoPage();
+        page.waitUntilPageIsReady();
+    }
+
+    @Test
+    void appEntryRedirectsToHomePage()
+    {
+        ApplicationsPanel applicationPanel = ApplicationsPanel.gotoPage();
+        ViewPage vp = applicationPanel.clickApplication("Admin Tools");
+        Assertions.assertEquals(AdminToolsHomePage.getSpace(), vp.getMetaDataValue("space"));
+        assertEquals(AdminToolsHomePage.getPage(), vp.getMetaDataValue("page"));
+    }
+
+    @Test
+    @Order(1)
+    void adminToolsHomePageBackend(TestConfiguration testConfiguration)
+    {
+        if (!isSupportedServer) {
+            return;
+        }
```

Comment:
This is not the right way. See https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-export/xwiki-platform-export-pdf/xwiki-platform-export-pdf-test/xwiki-platform-export-pdf-test-docker/src/test/it/org/xwiki/export/pdf/test/ui/PDFExportExecutionCondition.java . And you can also force a specific servlet container if you want with https://dev.xwiki.org/xwiki/bin/view/Community/Testing/DockerTesting/#HExample2:ChoosingContainer2BDB2BBrowserandversions .

Reply:
I don't want to force a specific container, as in the future multiple servlet engines will be supported. But I added the condition to check if the used container is currently supported.